aspnetcore/performance/rate-limit.md

```diff
@@ -0,0 +1,186 @@
+
+## Rate limiter algorithms
+
+The [`RateLimiterOptionsExtensions`](/dotnet/api/microsoft.aspnetcore.ratelimiting.ratelimiteroptionsextensions) class provides the following extension methods for rate limiting:
+
+* [Fixed window](#fixed)
+* [Sliding window](#slide)
+* [Token bucket](#token)
+* [Concurrency](#concur)
+
+<a name="fixed"></a>
+
+### Fixed window limiter
+
+The [`AddFixedWindowLimiter`](/dotnet/api/microsoft.aspnetcore.ratelimiting.ratelimiteroptionsextensions.addfixedwindowlimiter#microsoft-aspnetcore-ratelimiting-ratelimiteroptionsextensions-addfixedwindowlimiter(microsoft-aspnetcore-ratelimiting-ratelimiteroptions-system-string-system-threading-ratelimiting-fixedwindowratelimiteroptions)) method uses a fixed time window to limit requests. When the time window expires, a new time window starts and the request limit is reset.
+
+Consider the following code:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/middleware/rate-limit/WebRateLimitAuth/Program.cs" id="snippet_fixed":::
+
+The preceding code:
+* Calls [UseRateLimiter](/dotnet/api/microsoft.aspnetcore.builder.ratelimiterapplicationbuilderextensions.useratelimiter) to enable rate limiting.
+* Creates a fixed window limiter with a policy name of `"fixed"` and sets:
+* `permitLimit` to 4 and the time `window` to 12. A maximum of 4 requests per each 12-second window are allowed.
+* `queueProcessingOrder` to `QueueProcessingOrder.OldestFirst`.
+* `queueLimit` to 2.
+
+Apps should use [Configuration](xref:fundamentals/configuration/index) to set limiter options. The following code updates the preceding code using [`MyRateLimitOptions`](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/fundamentals/middleware/rate-limit/WebRateLimitAuth/Models/MyRateLimitOptions.cs) for configuration:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/middleware/rate-limit/WebRateLimitAuth/Program.cs" id="snippet_fixed2":::
+
+<a name="slide"></a>
+
+### Sliding window limiter
+
+A sliding window algorithm:
+
+* Is similar to the fixed window limiter but adds segments per window. The window slides one segment (window)/(segments per window).
+* Limits the requests for a window to `permitLimit` requests.
+* Each time window is divided in `n` segments per window.
+* Requests taken from the expired time segment one window back (`n` segments prior to the current segment), are added to the current segment. We refer to the most expired time segment one window back as the expired segment.  Consider the following table which shows a sliding window limiter with a 30-second window, 3 segments per window and a limit of 100 requests:
+
+* The top row and first column shows the time segment.
```

Comment:
Indentation would help here too

Reply:
See [Sliding window limiter](https://review.docs.microsoft.com/en-us/aspnet/core/performance/rate-limit?view=aspnetcore-7.0&branch=pr-en-us-26756#sliding-window-limiter)
I don't understand the benefit from deviating from the standard.